src/lib/repositories/impl/order_repository_impl.py

```diff
@@ -30,5 +34,23 @@ def update_by_id(self, order_id, order):
     def get_orders_to_process(self):
         orders = self.get_all()
 
-        orders_to_process = filter(lambda order: order.status == OrderStatus.NEW_ORDER, orders)
+        orders_to_process = filter(
+            lambda order: order.status == OrderStatus.NEW_ORDER, orders
+        )
         return list(orders_to_process)
+
+    def get_order_ingredients_by_order_id(self, order_id):
+
+        order_ingredients = []
+        order_details = self.order_detail_repository.get_by_order_id(order_id)
+
+        for order_detail in order_details:
```

Comment:
Try to map order details to an array of product ids.

Create a function in product ingredient repository that accepts a list of product ids and return an array of product ingredients for those product ids.

You don't need to multiply elements by quantity

Reply:
done